### PR TITLE
[FEATURE] synchronize plugin permissions with App Manager 

### PR DESCRIPTION
**Pull request by sidohaakma**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits
- [x] Added Feature/Fix to release notes

### CODE REVIEW

**Comment by sidohaakma** [comment]
test this please

File: molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java [hunk 1]
```diff
@@ -91,40 +111,59 @@ public void beforeMethod()
 		when(app.getAppConfig()).thenReturn("{'config': 'test'}");
 		ClassLoader classLoader = getClass().getClassLoader();
 		File testFile = new File(classLoader.getResource("test-resources/js/test.js").getFile());
-		String absoluteTestFileResourcePath = testFile.getPath().replace("js/test.js", "");;
+		String absoluteTestFileResourcePath = testFile.getPath().replace("js/test.js", "");
+		;
```
Comment (fdlk) [inline molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java]: typo

File: molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java [hunk 2]
```diff
@@ -140,28 +179,90 @@ public void testServeAppInactiveApp() throws Exception
...
-		}
-		finally
-		{
-			assertEquals(expectedMessage, "uri:uri");
-		}
+
+		mockMvc.perform(get(AppController.URI + "/uri/"))
+			   .andExpect(status().is4xxClientError())
+			   .andExpect(model().attribute("errorMessageResponse",
+					   ErrorMessageResponse.create("Access denied for inactive app at location /app/uri", "AM07")))
```
Comment (fdlk) [inline molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java]: I'd expect the style to match other exception messages, i.e.
> Access denied for inactive app at location '/app/uri'.

(But the PR doesn't contain this message.)

File: molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java [hunk 3]
```diff
@@ -91,40 +111,59 @@ public void beforeMethod()
...
+										 springExceptionHandler)
 								 .setLocaleResolver(localeResolver)
 								 .build();
 	}
 
+	@AfterMethod
+	public void afterMethod()
+	{
+		PluginIdentity pluginIdentity = new PluginIdentity("uri");
+		when(userPermissionEvaluator.hasPermission(pluginIdentity, VIEW_PLUGIN)).thenReturn(false);
```
Comment (fdlk) [inline molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java]: Shouldn't be necessary, the mocks get recreated in beforeMethod.

File: molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java [hunk 4]
```diff
@@ -91,40 +111,59 @@ public void beforeMethod()
...
 			   .andExpect(status().isOk())
 			   .andExpect(model().attribute("app", appResponse))
 			   .andExpect(model().attribute("baseUrl", "/test/path"))
 			   .andExpect(view().name("view-app"));
 	}
 
+	@Test
+	public void testServeAppNoPermissions() throws Exception
+	{
+		mockMvc.perform(get(AppController.URI + "/uri/")).andExpect(status().isUnauthorized());
```
Comment (fdlk) [inline molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java]: Could also check the error message response.

File: molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java [hunk 5]
```diff
@@ -91,40 +111,59 @@ public void beforeMethod()
...
+	}
+
 	@Test
 	public void testServeAppRedirectToApp() throws Exception
 	{
-		mockMvc.perform(get(AppController.URI + "/uri"))
-			   .andExpect(status().is3xxRedirection());
+		PluginIdentity pluginIdentity = new PluginIdentity("uri");
+		when(userPermissionEvaluator.hasPermission(pluginIdentity, VIEW_PLUGIN)).thenReturn(true);
+		mockMvc.perform(get(AppController.URI + "/uri")).andExpect(status().is3xxRedirection());
```
Comment (fdlk) [inline molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java]: Could also check the location you're redirected to. (But existing test code already doesn't do this).

**Pull request merged**